x-pack/plugins/observability_ai_assistant/public/components/chat/chat_prompt_editor.tsx

```diff
@@ -30,21 +32,31 @@ export function ChatPromptEditor({
   disabled,
   hidden,
   loading,
-  initialMessage,
+  initialRole,
+  initialFunctionCall,
+  initialContent,
   onChangeHeight,
   onSendTelemetry,
   onSubmit,
 }: ChatPromptEditorProps) {
   const containerRef = useRef<HTMLDivElement>(null);
 
-  const isFocusTrapEnabled = Boolean(initialMessage?.message);
+  const isFocusTrapEnabled = Boolean(initialContent || initialFunctionCall);
 
-  const [innerMessage, setInnerMessage] = useState<Message['message'] | undefined>(
-    initialMessage?.message
+  const [mode, setMode] = useState<'prompt' | 'function'>(
+    initialFunctionCall?.name ? 'function' : 'prompt'
   );
 
-  const [mode, setMode] = useState<'prompt' | 'function'>(
-    initialMessage?.message.function_call?.name ? 'function' : 'prompt'
+  const initialInnerMessage = initialRole
+    ? {
+        role: initialRole,
+        content: initialContent ?? '',
```

Comment:
I think you can actually just set this to an empty string here? The user is not expected to edit the content, only the function call.

Reply:
Probably want to change the argument to `onSubmit` callback from `Message` to `Message['function_call']`.

Reply:
> I think you can actually just set this to an empty string here? The user is not expected to edit the content, only the function call.

Not sure if that's true; the user might want to edit his earlier text based prompt, no?

Reply:
Oh right, this just edits the prompt by the user? Do we need function call then?

Reply:
> Probably want to change the argument to onSubmit callback from Message to Message['function_call'].

The component edits both user prompts and user function calls. 

`ChatPromptEditor` has two child components, `ChatPromptEditorFunction` for function editing and `ChatPromptEditorPrompt` for prompt editing

Reply:
got it 👍